example_package/README.md

```diff
@@ -0,0 +1,456 @@
+- **exit_wrong(Msg&&... msg)** -
+Kończ sprawdzanie z błędem i daje 0 punktów, chyba, że ustawiono partial_score.
+Zwraca `WRONG\n[msg]\n0\n` albo `OK\n[partial_score_msg]; [msg]\n[partial_score]\n` albo `OK\n[msg]\n[partial_score]\n`.
+
+Jeżeli w programie używamy `partial_score` to:
+
+- Trzymajmy zmienną mówiącą jaki jest wynik częściowy.
+- Trzymajmy zmienną mówiącą ile user może zdobyć punktów (do_zdobycia).
+- Używamy `exit_ok_with_score(do_zdobycia)` zamiast `exit_ok()`.
+
+#### **checker_test**
+
+oi.h udostępnia możliwość pisania testów do chekerki.
+Te testy są uruchamiane tylko lokalnie.
+Istnieją 2 sposoby ich pisania.
+Przykład obu z nich jest zaimplementowany w przykładowym `abcchk.cpp`.
+
+#### **InwerVerdict**
+
+oi.h udostępnia nam obiekt `inwer_verdict` klasy InwerVerdict.
+Używamy go jako strumień wyjścia, a mianowicie:
+`oi::inwer_verdict.[coś] << [msg]`.
+Gdzie `msg` to wiadomość którą chcemy pokazać przed zakończeniem.
+Natomiast `coś` to jedna z podanych opcji:
+
+- **exit_ok()** - Kończy program pomyślnie.
+- **exit_wrong()** - Kończy program z błędem.
+
+My będziemy używać tylko `oi::inwer_verdict.exit_ok() << [msg]`.
+Druga opcja jest używana systemowo i będziemy ją zgłaszać np. przez `oi_assert()` lub `oi::bug(Msg&&... msg)`.
+
+#### **bug**
+
+Wywołując `oi::bug(Msg&&... msg)`, program zakończy się niepowodzeniem.
+Wyświetli on wtedy podaną wiadomość/ci, w formacie `BUG: [msg]` i zakończy działanie z kodem 2.
+
+#### **oi_assert**
+
+Działa podobnie do zwykłego asserta.
+Wywołując `oi_assert(condition, msg...);`, sprawdzi założenie, a jak będzie błędne to poda dokładny komunikat co jest nie tak.
+Wypisze on `[FILE]:[LINE]: [func]: Assertion '[condition]' failed.` lub
+`[FILE]:[LINE]: [func]: Assertion '[condition]' failed: [msg]`
+
+#### **Random**
+
+Służy do losowania wartości i jest wymagane używać go zamiast zwykłego rand() lub std::mt19937 lub innych mechanizmów losujących.
+Zapewnia on, że liczby są rzeczywiście (pseudo) losowe.
+Klasa `Random` udostępnia:
+
+- **Random(uint_fast64_t seed = 5489)**
+- **void shuffle(T begin, T end)**
+- **void shuffle(T& container)**
+- **operator()(T min, T max)**
+
+Tak więc aby utworzyć obiekt robimy `oi::Random rng;` lub `oi::Random rng(seed);`.
+Aby zmienić seed nadpisujemy `rng = oi::Random(new_seed);`.
+Aby użyć robimy `rng(min, max);`.
+Pod wartości min i max podstawiamy zakres z jakiego chcemy wylosować wartość.
+Obsługiwane są wszystkie typy numeryczne (int, float, char, ...).
+Możemy również pomieszać jakiś kontener Używając `rng.shuffle()`, podając mu albo kontener albo początek i koniec przedziału.
+
+## in i out
+
+Są to foldery, w których znajdują się testy.
+Testy nazywamy `{TAG}{grupa}{nr_testu}.{in/out}`.
+
+Grupa:
+
+- 0 - są to testy wstępne, nie liczą się do oceny i uczestnik ma do nich dostęp na zawodach.
+- 1,2,... - zwyczajna grupa, punkty za nią dostaniemy jak przejdą wszystkie testy z danej grupy.
+
+nr_testu to kolejne litery alfabetu.
```

Comment:
to ciąg alfanumeryczny, zaczynający się od litery. 
Mam nadzieję że nie blefuję tutaj @A-dead-pixel ?

Reply:
potwierdzam